### PR TITLE
Visualizing data generated for a random walk using Python

### LINKED ISSUE #7247: Generating data for a random walk using Python
### Writing sample(s): 
https://www.section.io/engineering-education/visualizing-repositories-using-pygal/

### Topic Suggestion
Generating data for a random walk using Python

#### Proposed article introduction
 In this tutorial we’ll use Python to generate data for a random walk and then use matplotlib to create a visually appealing representation of the generated data. A random walk is a path that has no clear direction but is determined by a series of random decisions, each of which is left entirely to chance. You might imagine a random walk as the path an ant would take if 
it had lost its mind and took every step in a random direction. Random walks have practical applications in nature, physics, biology, chemistry, and economics. For example, a pollen grain floating on a drop of water moves across the surface of the water because it is constantly being pushed around by water molecules. Molecular motion in a water drop is random, so the path a pollen grain traces out on the surface is a random walk. The code we’ll write in this article models many real-world  situations

#### Key takeaways 

- Creating the RandomWalk() Class
- Choosing Directions
- Plotting the Random Walk
- Coloring the Points
- Plotting the Starting and Ending Points
- Adding Plot Points

#### Article quality
This tutorial will use broader examples that will help learners understand how to generate and visualize data  for a random walk using Python easily. The code we’ll write in this article models many real-world  situations. Also a different approach is taken to explain this tutorial making it different from other articles available over the internet.

### PR DESCRIPTION
**Pull request by duncandegwa**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [X] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [X] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [X] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #7247

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Comment by srishilesh** [comment]
Hey @BrianaNzivu ,

I have completed all the reviews in my backlog, and I'm waiting for the CM's final review. Just checking if I can take up this PR for review, since you have many in your backlog already.

Let me know.

Thanks.

**Comment by BrianaNzivu** [comment]
Hey @srishilesh .
Sure.

**Comment by srishilesh** [comment]
Hey @BrianaNzivu ,

Thank you.

File: content/articles/generating-data-for-a-random-walk-using-python/index.md
Comment (srishilesh) [inline content/articles/generating-data-for-a-random-walk-using-python/index.md:7]: Before we get started with the implementation, can you add a sub-topic that discusses where we would use this? Any application of it? How it is used?

Some readers might be new to this.

File: content/articles/generating-data-for-a-random-walk-using-python/index.md
Comment (srishilesh) [inline content/articles/generating-data-for-a-random-walk-using-python/index.md:9]: `Three characteristics are required for the class:`

Make the 3 characteristics as a numbered list.

Comment (srishilesh) [reply content/articles/generating-data-for-a-random-walk-using-python/index.md:9]: I find only 2 characteristics. Did you miss any?

File: content/articles/generating-data-for-a-random-walk-using-python/index.md
Comment (srishilesh) [inline content/articles/generating-data-for-a-random-walk-using-python/index.md:72]: Fix the indentation

File: content/articles/generating-data-for-a-random-walk-using-python/index.md
Comment (srishilesh) [inline content/articles/generating-data-for-a-random-walk-using-python/index.md:91]: Fix the indentation

File: content/articles/generating-data-for-a-random-walk-using-python/index.md
Comment (srishilesh) [inline content/articles/generating-data-for-a-random-walk-using-python/index.md:144]: Add a GitHub repo URL containing the source code

File: content/articles/generating-data-for-a-random-walk-using-python/index.md
Comment (srishilesh) [inline content/articles/generating-data-for-a-random-walk-using-python/index.md:145]: Add a `Further reading` section that contains additional references that the reader can learn from.

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @duncandegwa ,

Great work.

I have suggested some minor changes. Please ping me when you are done.

Thanks.

**Comment by duncandegwa** [comment]
Hey @srishilesh  
I have made the minor changes mentioned above
Thanks

**Comment by hectorkambow** [comment]
👏  

Nicely done!
@duncandegwa 

Also I'd recommend (if you have not already) to add a skill section to your author profile - that will go a long way with future potential employers.

**Pull request merged**
